src/main/webapp/view/discussion/discussion-comment.js

```diff
@@ -108,27 +110,36 @@ export default class DiscussionComment extends HTMLElement {
 
   /**
    * Posts the content of the reply textarea as a reply to this comment,
-   * and reloads the discussion area.
+   * and resets the reply form.
    */
   postReplyClicked() {
     const textarea = this.shadowRoot.querySelector(
         DiscussionComment.#SELECTOR_REPLY_TEXTAREA);
     this.#discussion.postReply(textarea.value, this.comment.commentKey.id);
+
+    textarea.value = '';
+    const replyForm =
+        this.shadowRoot.querySelector(DiscussionComment.#SELECTOR_REPLY_FORM);
+    $(replyForm).collapse('hide');
   }
 
   /**
-   * Creates a `DiscussionComment` for every reply to this comment, and
-   * adds them to a `<div>` in the replies slot of the DOM template.
+   * Inserts a new reply such that replies are sorted with the earliest created
+   * appearing first.
    */
-  addReplies(replies) {
-    const replyDiv = document.createElement('div');
-    replyDiv.slot = DiscussionComment.#SLOT_REPLIES;
-    for (const reply of replies) {
-      const child = new DiscussionComment(this.#discussion);
-      child.setComment(reply);
-      replyDiv.appendChild(child);
+  insertReply(newComment) {
+    const newCommentDate = new Date(newComment.created);
+    // For now, we use a linear search. This can be improved if it becomes
+    // an issue.
+    for (const commentElement of this.#replyDiv.children) {
+      const commentDate = new Date(commentElement.comment.created);
+      if (newCommentDate <= commentDate) {
+        commentElement.before(newComment.element);
+        return;
+      }
     }
-    this.appendChild(replyDiv);
+    // If it isn't before any existing replies, it must belong at the end.
+    this.#replyDiv.appendChild(newComment.element);
```

Comment:
Seems duplicated logic with insertRootComment. Can you refactor this code to create a shared method?

Reply:
Not easily.  they compare different things (timestamp vs creation date), and insert into different elements (private `replyDiv` vs static `ELEMENT_DISCUSSION`).

We talked offline and will keep as is